probatus/missing/imputation.py

```diff
@@ -0,0 +1,318 @@
+                Another option is using probatus.utils.Scorer to define a custom metric.
+
+            model_na_support(boolean): default True
+                If the classifier supports missing values by default e.g. LightGBM,XGBoost etc. 
+                If True an default comparison `Model Imputation` will be added indicating the results without any explict imputation. 
+                If False only the provided strategies will be used.
+            
+            n_jobs (int, optional):
+                Number of cores to run in parallel while fitting across folds. None means 1 unless in a
+                `joblib.parallel_backend` context. -1 means using all processors.
+
+            verbose (int, optional):
+                Controls verbosity of the output:
+
+                - 0 - nether prints nor warnings are shown
+                - 1 - 50 - only most important warnings regarding data properties are shown (excluding SHAP warnings)
+                - 51 - 100 - shows most important warnings, prints of the feature removal process
+                - above 100 - presents all prints and all warnings (including SHAP warnings).
+            
+            random_state (int, optional):
+                Random state set at each round of feature elimination. If it is None, the results will not be
+                reproducible and in random search at each iteration a different hyperparameters might be tested. For
+                reproducible results set it to integer.
+        """
+        self.clf = clf
+        self.model_na_support = model_na_support
+        self.scorer = get_single_scorer(scoring)
+        self.strategies = strategies
+        if cv is None:
+            self.cv = 5
+        else :
+            self.cv = cv
+        self.verbose = verbose
+
+        self.n_jobs = n_jobs
+        
+        if random_state is None:
+            self.random_state = 42
+        else:
+            self.random_state = random_state
+
+        self.fitted = False
+        self.report = pd.DataFrame([])
+
+    def __repr__(self):
+        return "Imputation comparision for {}".format(self.clf.__class__.__name__)
+
+
+    def fit(self, X, y,column_names=None,categorical_columns='auto'):
+        """
+        Calculates the cross validated results for various imputation strategies.
+        
+        Args:
+            X (pd.DataFrame):
+                input variables.
+
+            y (pd.Series):
+                target variable.
+
+            column_names (None, or list of str, optional):
+                List of feature names for the dataset. 
+                If None, then column names from the X dataframe are used.
+            
+            categorical_features (None, or list of str, optional):default=auto
+                List of categorical features.The imputation strategy for categorical 
+                is different that compared to numerical features. If auto try to infer
```

Comment:
I dont understand the docstring. 

I also don't really get the idea behind this param. Maybe better to use `drop_categorical` or something like this?

Reply:
Missing values in categorical columns are handled differently than for numerical values. For categorical columns we mark the missing values as `missing` and then apply Onehotencoding. Made it simpler by removing the parameter and auto detection and handling of categorical features.